src/services/VirtualFileSystemService.spec.ts

```diff
@@ -1,532 +1,423 @@
+        expect(existsDir).toEqual(true);
+      } finally {
+        await virtualFileSystemService.deleteDirectory(mockFolderToDelete);
+      }
     });
 
     it("throws an error if the parent directory doesn't exist", async () => {
-      await expect(service.createDirectory('/non/existing/file.ts')).rejects.toThrow('is not a directory');
+      const mockFilePath = path.resolve('/notExistingMockFolder/file.ts');
+
+      await expect(virtualFileSystemService.createDirectory(mockFilePath)).rejects.toThrow(/ENOENT: no such file or directory/);
     });
 
     it('throws an error if the parent directory is not a directory', async () => {
-      await expect(service.createDirectory('/src/index.ts/newDir')).rejects.toThrow('is not a directory');
+      const mockFilePath = path.resolve('/mockFolder/mockFile.ts/file.ts');
+
+      await expect(virtualFileSystemService.createDirectory(mockFilePath)).rejects.toThrow('No such directory');
     });
 
     it('throws an error if the target is a directory', async () => {
-      await expect(service.createDirectory('/src')).rejects.toThrow('Dir already exists');
+      const mockFilePath = path.resolve('/mockFolder');
+
+      await expect(virtualFileSystemService.createDirectory(mockFilePath)).rejects.toThrow(
+        "EEXIST: file already exists, mkdir '" + mockFilePath + "'",
+      );
     });
 
     it("throws an error if the target isn't a directory", async () => {
-      await expect(service.createDirectory('/src/index.ts')).rejects.toThrow('Dir already exists');
+      const mockFilePath = path.resolve('/mockFolder/mockFile.ts');
+
+      await expect(virtualFileSystemService.createDirectory(mockFilePath)).rejects.toThrow(
+        "EEXIST: file already exists, mkdir '" + mockFilePath + "'",
+      );
     });
 
-    it('throws an error if the target is a broken symbolic link', async () => {
-      await expect(service.createDirectory('/src/indexSLbroken.ts')).rejects.toThrow('Dir already exists');
+    it('throws an error if the target is a broken symbolc link', async () => {
+      const mockFilePath = path.resolve('/mockFolder/mockFileSLbroken.ln');
+
+      await expect(virtualFileSystemService.createDirectory(mockFilePath)).rejects.toThrow(
+        "EEXIST: file already exists, mkdir '" + mockFilePath + "'",
+      );
     });
   });
 
   describe('#deleteDirectory', () => {
     it('exists() returns false after deleting', async () => {
-      const dirToDelete = '/dirToDelete';
+      const mockFolderToDelete = path.resolve('/deleteDirTest');
+      let existsDir: boolean;
 
-      await service.createDirectory(dirToDelete);
-      let existsDir = await service.exists(dirToDelete);
-      expect(existsDir).toBe(true);
+      try {
+        await virtualFileSystemService.createDirectory(mockFolderToDelete);
 
-      await service.deleteDirectory(dirToDelete);
+        existsDir = await virtualFileSystemService.exists(mockFolderToDelete);
+        expect(existsDir).toEqual(true);
```

Comment:
Do we really need this? Shouldn't we trust `createDirectory`? I mean, we test `createDirectory` above...